datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/validate_profile.py

```diff
@@ -0,0 +1,146 @@
+        self.invalid = False
+        self.contents = ""
+
+    def __repr__(self):
+        return self.file_path
+
+
+def find_profiles(file, directory):
+    profiles_list = []
+    profile = Profile()
+    if file:
+        profile.file_path = file
+        profiles_list.append(profile)
+        return profiles_list
+
+    if directory:
+        profiles_list = get_all_profiles_from_dir(directory)
+        return profiles_list
+
+    else:
+        profiles_path = os.path.join(get_root(), "snmp", "datadog_checks", "snmp", "data", "profiles")
+        profiles_list = [f for f in os.listdir(profiles_path) if isfile(join(profiles_path, f))]
+        profiles_path = os.path.join(get_root(), "snmp", "datadog_checks", "snmp", "data", "profiles")
+        profiles_list = get_all_profiles_from_dir(profiles_path)
+        return profiles_list
+
+    return profiles_list
+
+
+def get_all_profiles_from_dir(directory):
+    profiles_list = []
+    profiles_path = directory
+    try:
+        dir_contents = [file for file in os.listdir(profiles_path) if isfile(join(profiles_path, file))]
+    except FileNotFoundError:
+        echo_failure("Directory not found, or could not be read")
+        abort()
+    for file in dir_contents:
+        profile = Profile()
+        profile.file_path = os.path.join(profiles_path, file)
+        profiles_list.append(profile)
+    return profiles_list
+
+
+def read_profile(profiles_list):
+    read_profiles = []
+    for profile in profiles_list:
+        try:
+            with open(profile.file_path) as f:
+                file_contents = yaml.safe_load(f.read())
+                profile.contents = file_contents
+        except OSError:
+            echo_failure("Profile file not found, or could not be read")
+            abort()
+        if not file_contents:
+            echo_failure("File contents returned None: " + str(profile))
+            abort()
+        read_profiles.append(profile)
+
+    return read_profiles
+
+
+def validate_with_jsonschema(profiles_list, verbose):
+    schema_file = os.path.join(
+        get_root(),
+        "datadog_checks_dev",
+        "datadog_checks",
+        "dev",
+        "tooling",
+        "commands",
+        "meta",
+        "snmp",
+        "snmp_profile.json",
+    )
+    with open(schema_file, "r") as f:
+        contents = f.read()
+        schema = json.loads(contents)
+    validator = jsonschema.Draft7Validator(schema)
+    for profile in profiles_list:
+        if verbose:
+            echo_info("Validating file: " + profile.file_path)
+        errors = validator.iter_errors(profile.contents)
+    for error in errors:
+        profile.errors.append(error)
+
+    # TODO - condition if there are no errors found
+    return profiles_list
+
+
+def produce_errors(profiles_list, verbose):
+    error_list = []
+    for profile in profiles_list:
+        if profile.errors:
+            error_list.append(profile)
+            profiles_list.remove(profile)
```

Comment:
changing the input could be unexpected to the function caller. you could clean up the profile list in a separate call, something like `remove_profiles_with_errors` and use this function to only print errors to console